src/lv_widgets/lv_win.c

```diff
@@ -669,4 +704,62 @@ static void lv_win_realign(lv_obj_t * win)
     lv_obj_align(ext->page, ext->header, LV_ALIGN_OUT_BOTTOM_LEFT, 0, 0);
 }
 
+static lv_obj_t * lv_win_btn_create(lv_obj_t * par, const lv_obj_t * copy)
+{
+	LV_LOG_TRACE("win btn create started");
+
+	lv_obj_t * win_btn;
+
+	win_btn = lv_btn_create(par, copy);
+    LV_ASSERT_MEM(win_btn);
+    if(win_btn == NULL) return NULL;
+
+    /*Allocate the extended data*/
+    lv_win_btn_ext_t * ext = lv_obj_allocate_ext_attr(win_btn, sizeof(lv_win_btn_ext_t));
+    LV_ASSERT_MEM(ext);
+    if(ext == NULL) {
+        lv_obj_del(win_btn);
+        return NULL;
+    }
+
+    if(ancestor_signal == NULL) ancestor_signal = lv_obj_get_signal_cb(win_btn);
+    if(ancestor_header_design == NULL) ancestor_header_design = lv_obj_get_design_cb(win_btn);
```

Comment:
These 2 lines re not required here.
They are used in `lv_win_create` to save some original signal functions and adding these lines here would overwrite them.